### PR TITLE
Update poltergeist to 1.8.1

### PR DESCRIPTION
**Pull request by steventux**
[Older versions of poltergeist do not always return the correct exit code](https://github.com/teampoltergeist/poltergeist/issues/497) and can lead to test suites failing but CI passing, this behaviour has been seen in [travel advice publisher branch builds](https://ci.dev.publishing.service.gov.uk/job/govuk_travel_advice_publisher_branches/107/console).

See https://github.com/alphagov/travel-advice-publisher/pull/98 for detail.

### CODE REVIEW

File: test/integration/simple_smart_answers_test.rb [hunk 1]
```diff
@@ -333,7 +333,7 @@ class SimpleSmartAnswersTest < ActionDispatch::IntegrationTest
         end
 
         # Ensure the edition param isn't added to the form URL as it needs to be in a hidden field.
-        assert_equal "/the-bridge-of-death/y", page.find('form')["action"]
+        assert_match /\/the-bridge-of-death\/y$/, page.find('form')["action"]
```
Comment (benilovj) [inline test/integration/simple_smart_answers_test.rb]: would this be a bit more readable if it were `%r{/the-bridge-of-death/y$}`?

Comment (benlovell) [inline test/integration/simple_smart_answers_test.rb]: Can we use `%r(...)` here, so we can avoid escaping those forward-slashes?

**Comment by benlovell** [comment]
Thanks ✨

**Comment by dsingleton** [comment]
:+1: :100:

**Pull request merged**